src/test/java/pl/javastart/streamstask/ExampleTest.java

```diff
@@ -1,6 +1,101 @@
+    public void shouldWomenHasSize0WhenEmptyList() {
+        Collection<User> women = streamsTask.findWomen(new ArrayList<>());
+        assertThat(women).isEmpty();
+    }
+
+    @Test
+    public void shouldWomenHasNames() {
+        Collection<User> women = streamsTask.findWomen(users);
+        softAssertions.assertThat(women)
+                .as("Women names")
+                .extracting(User::getName)
+                .containsExactlyInAnyOrder("Alicja", "Patrycja");
+    }
+
+    @Test
+    public void shouldWomenNamesHasEndsWithA() {
+        Collection<User> women = streamsTask.findWomen(users);
+        softAssertions.assertThat(women)
+                .as("Women names")
+                .allMatch(user -> user.getName().endsWith("a"));
+        softAssertions.assertAll();
+    }
+
+    @Test
+    public void testAverageMenAge() {
+        Double averageMenAge = streamsTask.averageMenAge(users);
+        assertThat(averageMenAge).isEqualTo(22.25);
+    }
+
+    @Test
+    public void shouldAverageMenAge0ForEmptyCollection() {
+        Double averageMenAge = streamsTask.averageMenAge(new ArrayList<>());
+        assertThat(averageMenAge).isEqualTo(0.0);
+    }
+
+    @Test
+    public void shouldGroupExpensesByUserIdReturnEmptyMapWhenUsersEmpty() {
+        Map<Long, List<Expense>> expensesByUserId = streamsTask.groupExpensesByUserId(new ArrayList<>(), expenses);
+        assertThat(expensesByUserId).isEmpty();
+    }
+
+    @Test
+    public void shouldThrowNullPointerExceptionWhenUsersNull() {
+        assertThatThrownBy(() -> streamsTask.groupExpensesByUserId(null, expenses))
+                .isInstanceOf(NullPointerException.class);
+    }
 
+    @Test
+    public void testGroupExpensesByUserId() {
+        Map<Long, List<Expense>> expensesByUserId = streamsTask.groupExpensesByUserId(users, expenses);
+        softAssertions.assertThat(expensesByUserId.get(1L)).hasSize(2);
+        softAssertions.assertThat(expensesByUserId.get(2L)).hasSize(3);
+        softAssertions.assertThat(expensesByUserId)
+                .allSatisfy((key, value) -> {
+                    if (key == 3L || key == 4L || key == 5L || key == 6L) {
+                        assertThat(value).isEmpty();
+                    }
+                });
+        softAssertions.assertAll();
+    }
 
+    @Test
+    public void testGroupExpensesByUser() {
+        Map<User, List<Expense>> expensesByUser = streamsTask.groupExpensesByUser(users, expenses);
+        softAssertions.assertThat(expensesByUser.get(users.get(0))).hasSize(2);
+        softAssertions.assertThat(expensesByUser.get(users.get(1))).hasSize(3);
+        softAssertions.assertThat(expensesByUser.get(users.get(2))).isEmpty();
+        softAssertions.assertThat(expensesByUser.get(users.get(3))).isEmpty();
+        softAssertions.assertThat(expensesByUser.get(users.get(4))).isEmpty();
+        softAssertions.assertThat(expensesByUser.get(users.get(5))).isEmpty();
+        softAssertions.assertAll();
+    }
```

Comment:
Zamień softAssertions na assertAll.
W code review wspominam o dodaniu asercji z rozmiarem mapy, ale to jednak nie ma sensu, bo dla pozostałych użytkowników będą puste listy wydatków